torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -289,6 +289,9 @@ def _run(cls, rank, test_name, file_name):
         self = cls(test_name)
         self.rank = rank
         self.file_name = file_name
+
+        if torch.cuda.device_count() != self.world_size:
```

Comment:
Thanks for the fix. Please correct me if I'm wrong. 

`self.world_size` would take `WORLD_SIZE` environment variable, https://github.com/pytorch/pytorch/blob/78de12f58854237dc5987e10bb4ab4fb33b2d01b/torch/testing/_internal/distributed/distributed_test.py#L317-L321
which is defined here for NCCL backend https://github.com/pytorch/pytorch/blob/78de12f58854237dc5987e10bb4ab4fb33b2d01b/test/run_test.py#L259-L263

However, `self.world_size` is not equal to `torch.cuda.device_count()` when number of GPU is greater than 3.

Reply:
Good catch!